on-premise-docker-registry/README.md

```diff
@@ -0,0 +1,191 @@
+
+>**NOTE:** In this sample, nginx as a reverse proxy is used to pull the images from an on-premise Docker registry. The setup relies on the Docker registry API v2 and proves the concept. This can be replaced with another reverse proxy or a custom implementation based on API and requirements.
+
+## Prerequisites
+
+- [SAP BTP, Kyma runtime instance](../prerequisites/#kyma)
+- [Docker](../prerequisites/#docker)
+- [GNU Make](https://www.gnu.org/software/make/)
+- [Kubernetes tooling](../prerequisites/#kubernetes)
+- [Cloud Connector on your machine or test system](../prerequisites/#sap-cloud-connector)
+- [Connectivity Proxy instance configured in Kyma](https://help.sap.com/docs/btp/sap-business-technology-platform/configure-sap-btp-connectivity-in-kyma-environment)
+- [OpenSSL](https://www.openssl.org/) or another similar tool to generate the certificates
+- [htpasswd](https://httpd.apache.org/docs/2.4/programs/htpasswd.html) installed
+
+## Setup
+
+The commands have been verified on OSX. However, it should be possible to adapt them for a windows laptop.
+
+### Environment variables
+
+- Export the following environment variables:
+
+```shell
+export KUBECONFIG=<path-to-kubeconfig>
+export NAMESPACE={kyma-namespace-used-for-this-sample}
+export CLUSTER_DOMAIN={kyma-cluster-domain}
+export REG_USER_NAME={docker-registry-user}
+export REG_USER_PASSWD={docker-registry-password}
+export EMAIL={your-email}
+```
+
+### On-premise Docker registry
+
+In this sample, we set up a simple Docker registry running on your machine. This can be replaced with a productive alternate deployed in your corporate network. The only requirement is that it should be reachable using Cloud Connector.
+
+- Generate the self-signed certificate for enabling HTTPS for the Docker registry. Be sure to specify the `CN` as `myregistry.kyma`
+
+```shell
+make generate-self-signed-cert
+```
+
+- Add the newly created certificate to your trust storage
+
+```shell
+make trust-self-signed-cert
+```
+
+- Generate the `htpasswd`. This is used for accessing the on-premise Docker registry.
+
+```shell
+make generate-htpasswd
+```
+
+- Start the Docker registry server
+
+```shell
+make start-docker-registry
+```
+
+- Add DNS entry to the `/etc/hosts` file
+
+```shell
+127.0.0.1 myregistry.kyma
+```
+
+- Configure Cloud Connector to access the on-premise Docker registry.
+  ![cc-config](assets/cc-config.png)
+
+## nginx as reverse proxy
+
+We use Nginx as a reverse proxy to forward the HTTP requests for pulling Docker images from the on-premise Docker registry using Connectivity Proxy and Cloud Connector.
+
+It is exposed as the `NodePort` service. This exposes the Service on the Kubernetes worker node on a port. Kubelet is a component running on each Kubernetes worker node. Among other tasks, it is responsible for pulling the Docker images.
```

Comment:
```suggestion
nginx is exposed as the `NodePort` service. This exposes the Service on the Kubernetes worker Node on a port. On each worker Node, [kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) runs as the component that pulls the Docker images (among other tasks).
```
- Why is "service" once written in lower case and once capitalized? Is this correct?
- Node is capitalized acc to our [guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/02-style-and-terminology.md#camelcase).
- kubelet is lower case according to their own docs.